src/Configuration/ConfigurationAnalyzer.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Configuration;
+
+final class ConfigurationAnalyzer
+{
+    /** @var array<string, mixed> */
+    private $config;
+
+    /**
+     * @param array<string, mixed> $arr
+     */
+    public static function fromArray(array $arr): self
+    {
+        return new self($arr);
+    }
+
+    /**
+     * @param array<string, mixed> $config
+     */
+    private function __construct(array $config)
+    {
+        $this->config = $config;
+    }
+
+    public function isCountingUseStatements(): bool
+    {
+        return (bool) ($this->config['count_use_statements'] ?? true);
```

Comment:
```suggestion
        return (bool) $this->config['count_use_statements'];
```
The null coalesce should not be necessary, because the Definition already sets this to `true`, if it's not set. I would like to avoid setting the default in multiple places to avoid bugs.

Reply:
At runtime, no, but at "compile" time it will make SA tools really angry as information of this character is not properly propagated throughout the codebase.

Reply:
I made the PHPDocs more restrictive in this file specifically to make it work in https://github.com/qossmic/deptrac/pull/571/commits/45ce6e08efd9ecaa4a722445c515b9bc65d62324